src/main/java/SolvabilityChecker/MazeSolver.java

```diff
@@ -0,0 +1,66 @@
+package SolvabilityChecker;
+
+import entities.DesignableMaze;
+import entities.Maze;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+public class MazeSolver {
+
+    private static List<int[]> getValidMoves(DesignableMaze maze, int position[]) {
+        // Intialize an empty arrayList to store the valid moves
+        List<int[]> valid_moves = new ArrayList<int[]>();
+        // Initialize an array to store the cell representations of moves in all 4 directions
+        int[][] possible_moves = {{position[0] + 1, position[1]}, {position[0] - 1, position[1]},
+                {position[0], position[1] + 1}, {position[0], position[1] - 1}};
+
+        // Add all valid moves in possible_moves to the valid_moves arrayList
+        for (int[] move: possible_moves) {
+            if (maze.inBounds(move[0], move[1]) && maze.getCell(move[0], move[1]) != Maze.ENCODING.get("wall")) {
+                valid_moves.add(move);
+            }
+        }
+
+        // Return all valid moves
+        return valid_moves;
+    }
+
+    public static boolean checkMazeSolvability(DesignableMaze maze) {
+        // Store the maze's start cell
+        int[] startLocation = maze.getStartLocation();
+        // Store the maze's end cell
+        int[] endLocation = maze.getEndLocation();
+        // Initialize an arrayList to store the cells to visit
+        List<int[]> queue = new ArrayList<int[]>() {
+            {add(startLocation);}
+        };
+        // Initialize a set to store all visited cells
+        Set<int[]> visited = new HashSet<int[]>() {
+            {add(startLocation);}
+        };
+
+        // Perform breadth first search as long as there are still cells to visit in the queue
+        while (queue.size() > 0) {
+            // Store the next cell to visit and remove it from the queue
+            int[] curr_position = queue.get(0);
+            queue.remove(0);
```

Comment:
slower than my grandma; this is an O(n) operation. 

for speed up us the java queue data structure or use a double linked list